src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py

```diff
@@ -780,6 +781,28 @@ def _get_kfp_client(
         client_args["host"] = self.config.kubeflow_hostname
         client_args["namespace"] = self.config.kubeflow_namespace
 
+        # Handle username and password, ignore the case if one is passed and not the other
+        # Also do not attempt to get cookie if cookie is already passed in client_args
+        if (
+            settings.client_username is not None
+            and settings.client_password is not None
+        ):
+            # If cookie is already set, then ignore
+            if "cookie" in client_args:
+                logger.warning(
+                    "Cookie already set in `client_args` so ignoring `client_username` and `client_password`..."
+                )
+            else:
+                logger.info(
```

Comment:
To clean this up a little, I would move this log and the one below into the `_get_session_cookie` method